### PR TITLE
Display separate "replays watched" tooltip for replays subsection

### LINKED ISSUE #12381: Wrong text is being used at Replay Watch History
In the profile page overlay, the "Replays Watched History" section, osu!lazer uses "Play" instead of "Replay Watched" text 

osu!lazer :
IMG_20210413_000853

web :
Screenshot_20210413_000831

### PR DESCRIPTION
**Pull request by frenzibyte**
Closes #12381

Also fixes `HistoryGraphTooltip` on each graph leaking each other's contents, due to no check present in `SetContent()` guarding against what graph is the given content part of.

### CODE REVIEW

File: osu.Game/Overlays/Profile/Sections/Historical/UserHistoryGraph.cs [hunk 1]
```diff
@@ -11,40 +11,46 @@ namespace osu.Game.Overlays.Profile.Sections.Historical
...
             public HistoryGraphTooltip(string tooltipCounterName)
                 : base(tooltipCounterName)
             {
+                this.tooltipCounterName = tooltipCounterName;
             }
 
             public override bool SetContent(object content)
             {
-                if (!(content is TooltipDisplayContent info))
+                if (!(content is TooltipDisplayContent info) || info.Name != tooltipCounterName)
```
Comment (peppy) [inline osu.Game/Overlays/Profile/Sections/Historical/UserHistoryGraph.cs:53]: Is there a reason this is required? Shouldn't it already update the name correctly below?

File: osu.Game/Overlays/Profile/UserGraph.cs [hunk 2]
```diff
@@ -208,6 +208,7 @@ private Vector2 calculateBallPosition(int index)
 
         protected abstract class UserGraphTooltip : VisibilityContainer, ITooltip
         {
+            protected new readonly OsuSpriteText Name;
```
Comment (peppy) [inline osu.Game/Overlays/Profile/UserGraph.cs:211]: Really don't like this being exposed as a `new`.. That's fine for tests but let's not do it in actual logic. Looks like you're only using the string portion of this anyway, so why not expose `tooltipCounterName` instead?

File: osu.Game/Overlays/Profile/Sections/Historical/UserHistoryGraph.cs (hunk 1 above)
Comment (frenzibyte) [reply osu.Game/Overlays/Profile/Sections/Historical/UserHistoryGraph.cs:53]: It doesn't update the name below? only the counter and bottom text (date)

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/Profile/UserGraph.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Overlays/Profile/UserGraph.cs:211]: It's not even being used anywhere as far as I can tell. Oversight maybe?

Comment (frenzibyte) [reply osu.Game/Overlays/Profile/UserGraph.cs:211]: Ah, is a drive-by change. Can be removed.

**Review by peppy — APPROVED** [review]
Seems fine

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**